From de86c9607f59b404981bd4a024c909ff42bfbf7f Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Fri, 9 Feb 2024 15:14:26 +0100
Subject: [PATCH 0944/1141] arm64: dts: st: move DSI node to stm32mp255

DSI support only on MP257 & MP255 SOCs.

Change-Id: Idc94de27b81efd1aad287634035301b808c8ec8b
Signed-off-by: Yannick Fertre <yannick.fertre@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/358878
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 arch/arm64/boot/dts/st/stm32mp251.dtsi | 14 --------------
 arch/arm64/boot/dts/st/stm32mp255.dtsi | 14 ++++++++++++++
 2 files changed, 14 insertions(+), 14 deletions(-)

diff --git a/arch/arm64/boot/dts/st/stm32mp251.dtsi b/arch/arm64/boot/dts/st/stm32mp251.dtsi
index a357891bc12b..b6c6f00c82e9 100644
--- a/arch/arm64/boot/dts/st/stm32mp251.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp251.dtsi
@@ -2060,20 +2060,6 @@ i3c4: i3c@46080000 {
 				status = "disabled";
 			};
 
-			dsi: dsi@48000000 {
-				compatible = "st,stm32mp25-dsi";
-				reg = <0x48000000 0x800>;
-				#clock-cells = <0>;
-				clocks = <&rcc CK_BUS_DSI>, <&rcc CK_KER_DSIPHY>,
-					 <&rcc CK_KER_LTDC>;
-				clock-names = "pclk", "ref", "px_clk";
-				resets = <&rcc DSI_R>;
-				reset-names = "apb";
-				feature-domains = <&rifsc STM32MP25_RIFSC_DSI_CMN_ID>;
-				power-domains = <&CLUSTER_PD>;
-				status = "disabled";
-			};
-
 			ltdc: display-controller@48010000 {
 				compatible = "st,stm32mp25-ltdc";
 				reg = <0x48010000 0x400>;
diff --git a/arch/arm64/boot/dts/st/stm32mp255.dtsi b/arch/arm64/boot/dts/st/stm32mp255.dtsi
index 1afcffd195da..1ea99d5102d8 100644
--- a/arch/arm64/boot/dts/st/stm32mp255.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp255.dtsi
@@ -8,6 +8,20 @@
 / {
 	soc@0 {
 		rifsc: rifsc@42080000 {
+			dsi: dsi@48000000 {
+				compatible = "st,stm32mp25-dsi";
+				reg = <0x48000000 0x800>;
+				#clock-cells = <0>;
+				clocks = <&rcc CK_BUS_DSI>, <&rcc CK_KER_DSIPHY>,
+					 <&rcc CK_KER_LTDC>;
+				clock-names = "pclk", "ref", "px_clk";
+				resets = <&rcc DSI_R>;
+				reset-names = "apb";
+				feature-domains = <&rifsc STM32MP25_RIFSC_DSI_CMN_ID>;
+				power-domains = <&CLUSTER_PD>;
+				status = "disabled";
+			};
+
 			vdec: vdec@480d0000 {
 				compatible = "st,stm32mp25-vdec";
 				reg = <0x480d0000 0x3c8>;
-- 
2.39.2

